pkg/cloud/scope/launchtemplate.go

```diff
@@ -0,0 +1,202 @@
+
+	AWSLaunchTemplate             *expinfrav1.AWSLaunchTemplate
+	MachinePool                   *expclusterv1.MachinePool
+	MachinePoolWithLaunchTemplate MachinePoolWithLaunchTemplate
+	InfraCluster                  EC2Scope
+	name                          string
+	additionalTags                infrav1.Tags
+}
+
+type MachinePoolWithLaunchTemplate interface {
+	GetObjectMeta() *metav1.ObjectMeta
+
+	GetSetter() conditions.Setter
+
+	PatchObject() error
+
+	GetLaunchTemplateIDStatus() string
+	SetLaunchTemplateIDStatus(id string)
+	GetLaunchTemplateLatestVersionStatus() string
+	SetLaunchTemplateLatestVersionStatus(version string)
+}
+
+type ResourceServiceToUpdate struct {
+	ResourceID      *string
+	ResourceService ResourceService
+}
+
+type ResourceService interface {
+	UpdateResourceTags(resourceID *string, create, remove map[string]string) error
+}
+
+// LaunchTemplateScopeParams defines a scope defined around a launch template.
+type LaunchTemplateScopeParams struct {
+	Client client.Client
+	Logger *logr.Logger
+
+	AWSLaunchTemplate             *expinfrav1.AWSLaunchTemplate
+	MachinePool                   *expclusterv1.MachinePool
+	MachinePoolWithLaunchTemplate MachinePoolWithLaunchTemplate
+	InfraCluster                  EC2Scope
+	Name                          string
+	AdditionalTags                infrav1.Tags
+}
+
+// NewLaunchTemplateScope creates a new LaunchTemplateScope from the supplied parameters.
+func NewLaunchTemplateScope(params LaunchTemplateScopeParams) (*LaunchTemplateScope, error) {
+	if params.Client == nil {
+		return nil, errors.New("client is required when creating a MachinePoolScope")
+	}
+	if params.MachinePool == nil {
+		return nil, errors.New("machinepool is required when creating a LaunchTemplateScope")
+	}
+	if params.InfraCluster == nil {
+		return nil, errors.New("cluster is required when creating a LaunchTemplateScope")
+	}
+
+	if params.Logger == nil {
+		log := klogr.New()
+		params.Logger = &log
+	}
+
+	return &LaunchTemplateScope{
+		Logger: *params.Logger,
+		Client: params.Client,
+
+		AWSLaunchTemplate:             params.AWSLaunchTemplate,
+		MachinePool:                   params.MachinePool,
+		MachinePoolWithLaunchTemplate: params.MachinePoolWithLaunchTemplate,
+		InfraCluster:                  params.InfraCluster,
+		name:                          params.Name,
+		additionalTags:                params.AdditionalTags,
+	}, nil
+}
+
+func (m *LaunchTemplateScope) IsEKSManaged() bool {
+	return m.InfraCluster.InfraCluster().GetObjectKind().GroupVersionKind().Kind == AWSManagedControlPlaneKind
+}
+
+func (m *LaunchTemplateScope) Name() string {
+	return m.name
+}
+
+func (m *LaunchTemplateScope) AdditionalTags() infrav1.Tags {
+	tags := make(infrav1.Tags)
+
+	// Start with the cluster-wide tags...
+	tags.Merge(m.InfraCluster.AdditionalTags())
+	// ... and merge in the Machine's
+	tags.Merge(m.additionalTags)
+
+	return tags
+}
+
+func (m *LaunchTemplateScope) GetRawBootstrapData() ([]byte, error) {
```

Comment:
We have this already on the [MachinePoolScope](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0babdf15d5584744c4693e8be4ab44c580847531/pkg/cloud/scope/machinepool.go#L126) so it feels like we should using that implementation and passing the result down into the service.

Reply:
Same with the tags above

Reply:
As a general pattern if we have the parent machine pool scope that kicks of reconciliation of another resource we can pass down the scope or the values instead of reimplemnting the functions.

Reply:
That makes sense. Updated the code to reuse implementation from MachinePoolScope